directories/src/redux/epics/node-epics.js

```diff
@@ -69,31 +66,23 @@ const requestBrokerEpic = (action$, store) => {
         });
         return Observable.fromPromise(
           iota.prepareTransfers({ address, message, value, tag, seed })
-        ).mergeMap(trytes => {
-          return Observable.fromPromise(
+        ).mergeMap(trytes =>
+          Observable.fromPromise(
             iota.attachToTangleCurl({
-              branchTransaction,
-              trunkTransaction,
+              branchTx,
+              trunkTx,
               mwm: 14,
               trytes
             })
-          ).mergeMap(trytesArray => {
-            return Observable.fromPromise(
-              brokerNode.completeBrokerNodeAddressPoW(txid, trytesArray)
-            ).flatMap(({ data }) => {
+          ).mergeMap(trytesArray =>
```

Comment:
Move this up 1 level (same for other chained mergeMaps), so its not so deeply nested when it doesn't have to be.

And always catch errors thrown in inner mergeMaps, it could take down the outer epic.

```js
.mergeMap(
   Observeable.fromPromise().catch()
)
.mergeMap()
```

Reply:
If I do it that way ^ won't it continue to the second `.mergeMap()`? I want the whole thing to fail at that point since all the steps are dependent of each other, if anything fails inbetween it should abandon the whole sequence.

Yeh it's a lot of nesting. I decided to go with this approach instead of chaining separate actions because
1. there would be a lot of forced argument passing since each call requires different things, so this is way easier to follow
2. they should all result in the same error if `.catch` is ever called (currently not implemented, it just calls Observable.empty)

Reply:
I didn't mean to chain separate actions (shouldn't do that), these should be atomic, just not nesting mergeMaps. You can pass vars to the next mergeMap. 

If it's too big of a refactoring, then let's not worry about it now. However, you should catch errors inside mergeMaps. If a stream encounter an error, it is shut down, so you cant catch it at the root epic, you need to catch it inside.

Reply:
>If I do it that way ^ won't it continue to the second .mergeMap()? I want the whole thing to fail at that point since all the steps are dependent of each other

It doesn't call the next one until the first observable pushes a value. It's the same behavior, just not nested